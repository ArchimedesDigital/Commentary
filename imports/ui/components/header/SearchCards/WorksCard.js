import { Card, CardHeader, CardText } from 'material-ui/Card';

// components:
import SearchTermButtonPanel from '/imports/ui/components/header/SearchTermButtonPanel'; // eslint-disable-line import/no-absolute-path

// helpers:
import { isActive, cardPropTypes, cardDefaultProps } from './helpers';


/*
	BEGIN WorksCard
*/
const WorksCard = ({ works, toggleWorkSearchTerm, styles, filters }) => (
	<Card
		className="search-tool-card"
	>
		<CardHeader
			title="Work"
			style={styles.cardHeader}
			actAsExpander
			showExpandableButton
			className="card-header"
		/>
		<CardText expandable style={styles.wrapper}>
			{works.map(work => (
				<SearchTermButtonPanel
					key={work._id}
					toggleSearchTerm={toggleWorkSearchTerm}
					label={work.title}
					searchTermKey="works"
					value={work}
					active={isActive(filters, work, 'works')}
				/>
			))}
		</CardText>
	</Card>
);
WorksCard.propTypes = {
	works: React.PropTypes.arrayOf(React.PropTypes.shape({
		_id: React.PropTypes.string.isRequired,
		title: React.PropTypes.string.isRequired,
		slug: React.PropTypes.string.isRequired,
	})),
	...cardPropTypes,
};
WorksCard.defaultProps = {
	works: [],
	...cardDefaultProps,
};
/*
	END WorksCard
*/

export default WorksCard;
