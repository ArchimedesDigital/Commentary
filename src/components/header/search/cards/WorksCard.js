import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardText } from 'material-ui/Card';

// components:
import SearchTermButtonPanel from '../SearchTermButtonPanel';

// helpers:
import { isActive } from './helpers';


/*
	BEGIN WorksCard
*/
const WorksCard = ({ works, toggleWorkSearchTerm, styles, filters }) => (
	<Card
		className="search-tool-card"
	>
		<CardHeader
			title="Work"
			style={styles.cardHeader}
			actAsExpander
			showExpandableButton
			className="card-header"
		/>
		<CardText expandable style={styles.wrapper}>
			{works.map(work => (
				<SearchTermButtonPanel
					key={work._id}
					toggleSearchTerm={toggleWorkSearchTerm}
					label={work.title}
					searchTermKey="works"
					value={work}
					active={isActive(filters, work, 'works')}
				/>
			))}
		</CardText>
	</Card>
);
WorksCard.propTypes = {
	works: PropTypes.arrayOf(PropTypes.shape({
		_id: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		slug: PropTypes.string.isRequired,
	})),
	//...cardPropTypes,
};
WorksCard.defaultProps = {
	works: [],
	//...cardDefaultProps,
};
/*
	END WorksCard
*/

export default WorksCard;
