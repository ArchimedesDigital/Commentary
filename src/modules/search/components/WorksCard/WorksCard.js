import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardText } from 'material-ui/Card';

// components:
import SearchTermButtonPanel from '../SearchTermButtonPanel';

// helpers:
import { isActive } from '../../lib/helpers';


/*
	BEGIN WorksCard
*/
const WorksCard = ({ works, toggleWorkSearchTerm, styles, filters }) => (
	<Card
		className="search-tool-card"
	>
		<CardHeader
			title="Work"
			style={styles.cardHeader}
			actAsExpander
			showExpandableButton
			className="card-header"
		/>
		<CardText expandable style={styles.wrapper}>
			{works.map(work => (
				<SearchTermButtonPanel
					key={work.id}
					toggleSearchTerm={toggleWorkSearchTerm}
					label={work.english_title}
					searchTermKey="works"
					value={work}
					active={isActive(filters, work, 'works')}
				/>
			))}
		</CardText>
	</Card>
);
WorksCard.propTypes = {
	works: PropTypes.arrayOf(PropTypes.shape({
		_id: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		slug: PropTypes.string.isRequired,
	})),
	//...cardPropTypes,
};
WorksCard.defaultProps = {
	works: [],
	//...cardDefaultProps,
};
/*
	END WorksCard
*/

export default WorksCard;
