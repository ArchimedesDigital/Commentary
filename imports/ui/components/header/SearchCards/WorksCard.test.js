import React from 'react';
import renderer from 'react-test-renderer';


// component:
import WorksCard from './WorksCard';


describe('WorksCard', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<WorksCard cardHeader="test" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

