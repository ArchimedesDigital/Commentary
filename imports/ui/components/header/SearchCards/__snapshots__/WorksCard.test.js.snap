// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorksCard renders correctly 1`] = `
<div
  className="search-tool-card"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      className="card-header"
      onTouchTap={[Function]}
      style={
        Object {
          "boxSizing": "border-box",
          "cursor": "pointer",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          Work
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        />
      </div>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "bottom": 0,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": "auto",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "absolute",
            "right": 4,
            "textDecoration": "none",
            "top": 0,
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;
